var/spack/repos/builtin/packages/util-macros/package.py

```diff
@@ -17,7 +17,7 @@ class UtilMacros(AutotoolsPackage, XorgPackage):
     # note: url_for_version can only return a single url, no mirrors
     def url_for_version(self, version):
         if self.spec.satisfies("@:1.19"):
-            return self.urls[0].replace("xz", "bz2")
+            return spack.url.substitute_version(self.urls[0].replace("xz", "bz2"), version)
```

Comment:
This method should be moved to below the directives, say around line 32.

Reply:
🤔 This does not appear to work with `spack checksum`.  I can only confirm `1.20.1` and `spack checksum util-macros` only produces checksums for that and `1.20.0`.